gcloud/tests/taskflow3/test_api.py

```diff
@@ -0,0 +1,196 @@
+            'name': 'first_task',
+            'incoming': TEST_ID_LIST[5],
+            'outgoing': TEST_ID_LIST[6],
+            'optional': True,
+            'component': {
+                'code': 'test',
+                'data': {
+                    'input_test': {
+                        'hook': False,
+                        'value': '${custom_key1}',
+                    },
+                    'radio_test': {
+                        'hook': False,
+                        'value': '1',
+                    },
+                },
+            }
+        },
+        TEST_ID_LIST[4]: {
+            'id': TEST_ID_LIST[4],
+            'type': 'ServiceActivity',
+            'name': 'first_task',
+            'incoming': TEST_ID_LIST[6],
+            'outgoing': TEST_ID_LIST[7],
+            'optional': True,
+            'component': {
+                'code': 'test',
+                'data': {
+                    'input_test': {
+                        'hook': True,
+                        'value': '${custom_key2}'
+                    },
+                    'radio_test': {
+                        'hook': False,
+                        'value': '2'
+                    },
+                },
+            }
+        },
+    },
+    'flows': {  # 存放该 Pipeline 中所有的线
+        TEST_ID_LIST[5]: {
+            'id': TEST_ID_LIST[5],
+            'source': TEST_ID_LIST[1],
+            'target': TEST_ID_LIST[3]
+        },
+        TEST_ID_LIST[6]: {
+            'id': TEST_ID_LIST[6],
+            'source': TEST_ID_LIST[3],
+            'target': TEST_ID_LIST[4]
+        },
+        TEST_ID_LIST[7]: {
+            'id': TEST_ID_LIST[7],
+            'source': TEST_ID_LIST[4],
+            'target': TEST_ID_LIST[2]
+        },
+    },
+    'gateways': {  # 这里存放着网关的详细信息
+    },
+    'constants': {
+        '${custom_key1}': {
+            'index': 0,
+            'name': 'input1',
+            'key': '${custom_key1}',
+            'desc': '',
+            'validation': '^.*$',
+            'show_type': 'show',
+            'value': 'value1',
+            'source_type': 'custom',
+            'source_tag': '',
+            'source_info': {},
+            'custom_type': 'input',
+        },
+        '${custom_key2}': {
+            'index': 1,
+            'name': 'input2',
+            'key': '${custom_key2}',
+            'desc': '',
+            'validation': '^.*$',
+            'show_type': 'show',
+            'value': 'value1',
+            'source_type': 'custom',
+            'source_tag': '',
+            'source_info': {},
+            'custom_type': 'input',
+        },
+    },
+    'outputs': ['${custom_key1}'],
+}
+
+
+class APITest(TestCase):
+
+    def setUp(self):
+        self.PREVIEW_TASK_TREE_URL = '/taskflow/api/preview_task_tree/{biz_cc_id}/'
+        self.client = Client()
+
+    @mock.patch('gcloud.taskflow3.api.JsonResponse', MockJsonResponse())
+    def test_preview_task_tree_constants_not_referred(self):
```

Comment:
`test_preview_task_tree_constants_not_referred` to `test_preview_task_tree__constants_not_referred`